pkg/agent/manager/cache/lru_cache_test.go

```diff
@@ -20,70 +20,211 @@ import (
+		print(a)
+
+		// Assert that reg0 is still returned, and reg1, reg3 and reg6 are not
+		assert.Equal(t, &WorkloadUpdate{
+			Identities:       []Identity{{Entry: reg0}, {Entry: reg2}, {Entry: reg4}, {Entry: reg5}},
+			Bundle:           bundleV1,
+			FederatedBundles: map[spiffeid.TrustDomain]*Bundle{},
 		},
-	}, workloadUpdate)
+			cache.FetchWorkloadUpdate(makeSelectors("A", "B", "C", "D", "E", "F", "G")))
+	})
+
+	t.Run("return the new identity after the first conflicting is deleted", func(t *testing.T) {
+		updateEntries := &UpdateEntries{
+			Bundles:             makeBundles(bundleV1),
+			RegistrationEntries: makeRegistrationEntries(reg1, reg2, reg3, reg4, reg5, reg6),
+		}
+		cache.UpdateEntries(updateEntries, nil)
+
+		updateSVIDs = &UpdateSVIDs{
+			X509SVIDs: makeX509SVIDs(reg1, reg2, reg3, reg4, reg5, reg6),
+		}
+		cache.UpdateSVIDs(updateSVIDs)
+
+		// Assert that reg1 is now returned and reg0, reg3 and reg6 are not
+		assert.Equal(t, &WorkloadUpdate{
+			Identities:       []Identity{{Entry: reg1}, {Entry: reg2}, {Entry: reg4}, {Entry: reg5}},
+			Bundle:           bundleV1,
+			FederatedBundles: map[spiffeid.TrustDomain]*Bundle{},
+		},
+			cache.FetchWorkloadUpdate(makeSelectors("A", "B", "C", "D", "E", "F", "G")))
+	})
 }
 
 func TestLRUCacheMatchingRegistrationIdentities(t *testing.T) {
 	cache := newTestLRUCache(t)
 
-	// populate the cache with FOO and BAR without SVIDS
-	foo := makeRegistrationEntry("FOO", "A")
-	bar := makeRegistrationEntry("BAR", "B")
-	updateEntries := &UpdateEntries{
-		Bundles:             makeBundles(bundleV1),
-		RegistrationEntries: makeRegistrationEntries(foo, bar),
-	}
-	cache.UpdateEntries(updateEntries, nil)
+	// populate the cache with entries without SVIDS
+	reg0 := makeRegistrationEntry("REG0", time.Now(), "internal", "A")
+	reg1 := makeRegistrationEntry("REG1", time.Now(), "internal", "B")
+	reg2 := makeRegistrationEntry("REG2", time.Now().Add(2*time.Hour), "external", "C")
+	reg3 := makeRegistrationEntry("REG3", time.Now().Add(3*time.Hour), "internal", "D")
+	reg4 := makeRegistrationEntry("REG4", time.Now().Add(4*time.Hour), "", "E")
+	reg5 := makeRegistrationEntry("REG5", time.Now().Add(5*time.Hour), "", "F")
+
+	t.Run("retrieve entries with no svids", func(t *testing.T) {
+		updateEntries := &UpdateEntries{
+			Bundles:             makeBundles(bundleV1),
+			RegistrationEntries: makeRegistrationEntries(reg1, reg2, reg3, reg4, reg5, reg0),
+		}
+		cache.UpdateEntries(updateEntries, nil)
 
-	assert.Equal(t, []*common.RegistrationEntry{bar, foo},
-		cache.MatchingRegistrationEntries(makeSelectors("A", "B")))
+		assert.Equal(t, []*common.RegistrationEntry{reg0, reg2, reg4, reg5},
+			cache.MatchingRegistrationEntries(makeSelectors("A", "B", "C", "D", "E", "F", "G")))
+	})
 
-	// Update SVIDs and MatchingRegistrationEntries should return both entries
+	// Update SVIDs and MatchingRegistrationEntries should now return entries
```

Comment:
This wording of this comment change is kind of confusing to me because the entries are also being returned in the previous assertion.

```suggestion
	// Update SVIDs and MatchingRegistrationEntries should return entries
```

Reply:
Yeah, the behavior on lru_cache is different, I've copied this comment from cache_test. I'm going to update it to correctly describe the behavior we're testing.